### DISCUSSION TITLE
Timed out while reading client stream prefix

### DISCUSSION THREAD

**Discussion by metanovii** [linkerd/linkerd2#5184] [General] [answered]
Hi! I'm trying to send a request, after which the service connects to an external database and then I get a response mysql gona away.
And log from proxy:
```
INFO ThreadId(01) outbound:accept{peer.addr=10.244.0.47:47582 target.addr={{ external_mysql_server_ip }}:3315}: linkerd2_app_core::serve: Connection closed error=Timed out while reading client stream prefix
```
what does it mean?

**Comment by grampelberg** [comment, marked as answer]
See [protocol detection](https://linkerd.io/2/features/protocol-detection/). MySQL is a server-speaks-first protocol, so you'll want to skip that port on the outbound.
